tutorial/creating_packages/other_types_of_packages/tool_requires_packages.rst

```diff
@@ -1,2 +1,139 @@
 Tool requires packages
 ======================
+
+In the ":ref:`Using build tools as Conan packages <consuming_packages_tool_requires>`" section we learned how to use
+a tool require to build (or help building) our project or Conan package.
+In this section we are going to learn how to create a recipe for a tool require.
+
+Please, first clone the sources to recreate this project. You can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: bash
+
+    $ git clone https://github.com/conan-io/examples2.git
+    $ cd examples2/tutorial/creating_packages/other_packages/tool_requires
+
+
+A simple tool require recipe
+----------------------------
+
+Let's see a recipe that even being very simple, covers most of the ``tool-require`` use-cases:
+
+.. code-block:: python
+    :caption: conanfile.py
+
+    import os
+    import stat
+    from conan import ConanFile
+    from conan.tools.files import save
+
+
+    class my_tool(ConanFile):
+        name = "my_tool"
+        version = "1.0"
+        package_type = "application"
+
+        settings = "os", "arch"
+
+        def package(self):
+            extension = "sh" if self.settings_build.os != "Windows" else "bat"
+            app_path = os.path.join(self.package_folder, "bin", "say_hello.{}".format(extension))
```

Comment:
I think I would prefer to see a more close to reality example here, with an executable... maybe some tool that verifies something in the consumer build? Because this could be closer to the typicall question that sometines users ask on "how to package all their build scripts and use them in another build?" we may put in a separate example

Reply:
I also want to see the case of the executable, I suggested that above, as the possible following example.
But I don't dislike this one either, I think it is a super common use case. We know tons of users that are packaging scripts, cmake utilities and other stuff in ``build_requires``. Maybe even more common that packaging executable tools?

Reply:
As I see it, I would explain the case of an executable in the main tutorial history (something inspired in tools like cmake, meson, ninja, nasm... ) and then the case of packaging build scripts in an example. If we want to show the case of the build scripts here in the main tutorial then I would make it a bit closer to the usual case exporting the scripts, not generating them... and make the title more close to: "How to package my build scripts to use in other recipes" that is the question that people will search for.

Reply:
Yes, that makes sense too, the case of using exported scripts, I agree, I already commented that.

I'd probably keep the "package my build script" here in the main tutorial line, not in an example, but I am fine either way.

Reply:
This is not trying to show how to package "build shell scripts" but the concept of packaging any application to be run in build. But as it is confusing you I'll try to change it. Before you suggest, I'll not use the ``cmake_exe`` template because the readers will learn nothing.